core/src/main/scala/org/apache/spark/sql/extensions/TiDDLRule.scala

```diff
@@ -0,0 +1,27 @@
+package org.apache.spark.sql.extensions
+
+import org.apache.spark.sql.{SparkSession, TiContext}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.command._
+
+case class TiDDLRule(getOrCreateTiContext: SparkSession => TiContext)(sparkSession: SparkSession)
+    extends Rule[LogicalPlan] {
+  protected lazy val tiContext: TiContext = getOrCreateTiContext(sparkSession)
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
+    // TODO: support other commands that may concern TiSpark catalog.
+    case ShowDatabasesCommand(databasePattern) =>
```

Comment:
Please check if any database searching related behavior consistent.
If we issue a drop database in TiDB, ideally it should print "drop table not supported". But here it looks like it would be "no such database".

Reply:
I will create an issue for this. For now, DDLs such as `drop database/table` are not supported.

Reply:
Currently, all unsupported operations will throw a NotImplementedError